### PR TITLE
Refactor create and edit event flow

### PR DESCRIPTION
**Pull request by kumarpalsinh25**
### **Issues reference:**
- fixes, #1436 
- fixes, #1437 
- ref, #1128
- - - - - -
### **Included Feature:** 
- Complete refactoring of the create and edit event flow
- Used flutter's default date and time picker instead of any third-part plugins
- Added AppFlowy HTML Editor for event description
- Added certain validation of date and time selection from flutter side itself to avoid false data passing to the server
    - Start date should not be past date
    - End date should not be less than start date
    - If start date and end date are same then end time should not be less than start date
    - Upon selection of start date, default end date automatically selected 
    - Upon selection of start time, default end time automatically selected with 1 hour of difference
    - And some other miscellaneous validations

- - - - - -
### **Reference Videos:** 
Mobile App:
https://github.com/acterglobal/a3/assets/51526480/15aab2d2-dc89-4bca-b65a-ec404785d957

Desktop App:
https://github.com/acterglobal/a3/assets/51526480/e8c28e72-5638-45be-8db8-2a2c9075fd8e

### CODE REVIEW

File: app/lib/features/events/pages/create_edit_event_page.dart [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+  final _endTimeController = TextEditingController();
+  DateTime _selectedEndDate = DateTime.now();
+  TimeOfDay _selectedEndTime = TimeOfDay.now();
+  EditorState textEditorState = EditorState.blank();
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
+      // if calendarId is not null that means Edit Event
```
Comment (gtalha07) [inline app/lib/features/events/pages/create_edit_event_page.dart:53]: Nice, I am now thinking similar about having one page that deals with create edit functions of pins.

**Review by gtalha07 — APPROVED** [review]
Looks good!

**Comment by gtalha07** [comment]
Reminder to add changelogs if meaningful changes for end-user!

File: app/lib/features/events/pages/create_edit_event_page.dart (hunk 1 above)
Comment (kumarpalsinh25) [reply app/lib/features/events/pages/create_edit_event_page.dart:53]: Yes, one file for create and edit flow is nice but based on the below conditions.
- Code complexity doesn't gets increased with common create and edit flow.
- When there are chances of high amount of code duplication.

**Pull request merged**